### PR TITLE
Fixed: Key signature changes

### LINKED ISSUE #525: Harmonize key.sig et al. attributes on scoreDef, staffDef, key and keySig
Since `staffDef@key.sig.showchange` is changing to `staffDef@keysig.showchange` in MEI 4, it is probably best to change `staffDef@key.sig` to `staffDef@keysig` for consistency.

There are `<key>` and `<keySig>` elements:

http://music-encoding.org/guidelines/v3/elements/key.html
http://music-encoding.org/guidelines/v3/elements/keysig.html

`<key>` should be used for analytical purposes, and `<keySig>` should be for print purposes, so it is best that `staffDef@key.sig.showchange` is changed to `staffDef@keysig.showChange` since it refers to the printed key signature.  Likewise `staffDef@key.sig` should change to `staffDef@keysig` for the same reason.

`<keySig>` has `@sig.showchange`.  Perhaps this should be changed to `@showchange` to also maintain a uniform style?

http://music-encoding.org/guidelines/v3/elements/keysig.html

### PR DESCRIPTION
**Pull request by ahankinson**
This PR contains changes to the key signature attributes on `<keySig>` and `<staffDef> / <scoreDef>`. It clarifies how key signature changes should be drawn, and makes the naming of the attributes more consistent.

Fixes #525

### CODE REVIEW

**Comment by ahankinson** [comment]
This is marked draft until I can write a paragraph or two in the guidelines about key signature changes.

File: source/modules/MEI.visual.xml [hunk 1]
```diff
@@ -824,10 +824,10 @@
     <desc xml:lang="en">Used by staffDef and scoreDef to provide default values for attributes in the visual
       domain related to key signatures.</desc>
     <attList>
-      <attDef ident="keysig.showchange" usage="opt">
+      <attDef ident="keysig.cancelaccid" usage="opt">
```
Comment (rettinghaus) [inline source/modules/MEI.visual.xml:827]: The description should be updated, as it is no longer only if, but rather where.

File: source/modules/MEI.xml [hunk 2]
```diff
@@ -510,6 +510,25 @@
...
+        <valItem ident="none">
+          <desc xml:lang="en">Do not show cancellation accidentals.</desc>
+        </valItem>
+        <valItem ident="before">
+          <desc xml:lang="en">Show cancellation accidentals before the new key accidentals.</desc>
+        </valItem>
+        <valItem ident="after">
+          <desc xml:lang="en">Show cancellation accidentals after the new key accidentals ("Old style" or "French")</desc>
+        </valItem>
+        <valItem ident="before-bar">
```
Comment (rettinghaus) [inline source/modules/MEI.xml:526]: I don't think this value makes much sense here, as `keySig` is a layer element and does not have do appear next to a measure.

Random example from Ravel's _Le tombeau de Couperin_

Le tombeau de Couperin (excerpt)

If you need to split it up, you should encode the cancellation at the end of one measure and the new one at the beginning of the next. (Maybe connect them with a linking attribute like `@precedes` and `@follows`). 

Or what to do if you have a `barLine` element next to it, but in the following order: 

```xml
<note />
<keySig keysig.cancelaccid="before-bar" keysig="3f" />
<barLine />
<note />
```

File: source/modules/MEI.xml [hunk 3]
```diff
@@ -510,6 +510,25 @@
       </valList>
     </content>
   </macroSpec>
+  <macroSpec ident="data.CANCELACCID" module="MEI" type="dt">
+    <desc xml:lang="en">Indicates the location of the tonic in the circle of fifths.</desc>
+    <content>
+      <valList type="closed">
+        <valItem ident="none">
+          <desc xml:lang="en">Do not show cancellation accidentals.</desc>
```
Comment (rettinghaus) [inline source/modules/MEI.xml:518]: Maybe better to take the encoder's perspective than the renderer's: "Without cancellation accidentals."? Possibly with a "Written" or "Printed" in front of that.

**Review by rettinghaus — CHANGES_REQUESTED** [review]
Thanks, mostly looking fine.

File: source/modules/MEI.visual.xml (hunk 1 above)
Comment (ahankinson) [reply source/modules/MEI.visual.xml:827]: Fixed! Thanks.

File: source/modules/MEI.xml [hunk 4]
```diff
@@ -510,6 +510,25 @@
       </valList>
     </content>
   </macroSpec>
+  <macroSpec ident="data.CANCELACCID" module="MEI" type="dt">
+    <desc xml:lang="en">Indicates the location of the tonic in the circle of fifths.</desc>
```
Comment (ahankinson) [inline source/modules/MEI.xml]:
```suggestion
    <desc xml:lang="en">Indicates where cancellation accidentals are shown in a key signature.</desc>
```

File: source/modules/MEI.xml (hunk 3 above)
Comment (ahankinson) [reply source/modules/MEI.xml:518]: I think "Do not show..." is clearer, as it aligns with the other definitions that all say "Show ...".

File: source/modules/MEI.xml (hunk 2 above)
Comment (ahankinson) [reply source/modules/MEI.xml:526]: (Sorry, I had written this response before I resolved it, but it didn't save for some reason.)

> Random example from Ravel's Le tombeau de Couperin

You can still do this:

`<keySig keysig.cancelaccid="after" ... />`

We make no guarantees in MEI that all values are valid in all circumstances.

File: source/modules/MEI.xml (hunk 3 above)
Comment (rettinghaus) [reply source/modules/MEI.xml:518]: I meant that for all descriptions.

File: source/modules/MEI.xml [hunk 5]
```diff
@@ -510,6 +510,25 @@
...
+          <desc xml:lang="en">Do not show cancellation accidentals.</desc>
+        </valItem>
+        <valItem ident="before">
+          <desc xml:lang="en">Show cancellation accidentals before the new key accidentals.</desc>
+        </valItem>
+        <valItem ident="after">
+          <desc xml:lang="en">Show cancellation accidentals after the new key accidentals ("Old style" or "French")</desc>
+        </valItem>
+        <valItem ident="before-bar">
+          <desc xml:lang="en">Show cancellation accidentals before the barline.</desc>
```
Comment (lpugin) [inline source/modules/MEI.xml]:
```suggestion
          <desc xml:lang="en">Show cancellation accidentals before the barline (also known as "Russian").</desc>
```

File: source/docs/04-cmn.xml [hunk 6]
```diff
@@ -231,8 +231,8 @@
                   </p>
                </div>
                <div xml:id="cmnReDef" type="div3">
-                  <head>Redefinition of Score Parameters</head>
-                  <p>Sometimes it is necessary to provide the parameters of a score or a staff with new values. For example. a score may change keys, gain or lose staves, use different layout settings at any point, etc. Likewise, a staff may change its clef, gain or lose layers, or become invisible, and so on. To accommodate these circumstances, in CMN <gi scheme="MEI">staffDef</gi> is allowed to occur in the following locations:</p>
+                  <head>Re-definition of Score Parameters</head>
+                  <p>Sometimes it is necessary to re-define the parameters of a score or a staff. For example. a score may change keys, the number of staves, or use different layout settings. Likewise, a staff may change its clef, change the number of layers, or become invisible. To accommodate these changes, <gi scheme="MEI">staffDef</gi> is allowed to occur in the following locations:</p>
```
Comment (rettinghaus) [inline source/docs/04-cmn.xml]:
```suggestion
                  <p>Sometimes it is necessary to re-define the parameters of a score or a staff. For example, a score may change keys, the number of staves, or use different layout settings. Likewise, a staff may change its clef, change the number of layers, or become invisible. To accommodate these changes, <gi scheme="MEI">staffDef</gi> is allowed to occur in the following locations:</p>
```

File: source/modules/MEI.xml (hunk 2 above)
Comment (rettinghaus) [reply source/modules/MEI.xml:526]: At least there should be a scheamtron warning when using this value while `keySig` is a `descendant` of measure.

**Review by rettinghaus — CHANGES_REQUESTED** [review]

File: source/docs/04-cmn.xml [hunk 7]
```diff
@@ -244,7 +244,14 @@
                      <item>within sections and endings; that is, inside <gi scheme="MEI">section</gi> and <gi scheme="MEI">ending</gi> elements; and</item>
                      <item>between sections and endings; that is, directly within a <gi scheme="MEI">score</gi> or <gi scheme="MEI">part</gi>.</item>
                   </list>
-                  <p>The possibility also exists to include <gi scheme="MEI">scoreDef</gi> and <gi scheme="MEI">staffDef</gi> in staves and layers when the mei-all schema is in use; however, this practice is not recommended for the CMN repertoire.</p>
+                  <p>It is also possible to include <gi scheme="MEI">scoreDef</gi> and <gi scheme="MEI">staffDef</gi> in staves and layers when the MEI All schema is in use; however, this practice is not recommended for the CMN repertoire.</p>
+                  <p>The following example shows how to change the key and meter signatures within a score. The <att>keysig.cancelaccid</att> attribute may be used to control the position of the cancellation accidentals, while the <att>keysig.visible</att> can be used to hide the key signature entirely.</p>
```
Comment (musicEnfanthen) [inline source/docs/04-cmn.xml]:
```suggestion
                  <p>The following example shows how to change the key and meter signatures within a score. The <att>keysig.cancelaccid</att> attribute may be used to control the position of the cancellation accidentals of the key signature change, while the <att>keysig.visible</att> can be used to hide the key signature entirely.</p>  
```

Comment (musicEnfanthen) [reply source/docs/04-cmn.xml]: Suggested the clarification, since the last reference was to "key and meter signatures".

**Review by lpugin — APPROVED** [review]

**Review by bwbohl — APPROVED** [review]

**Pull request merged**